lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -1067,6 +1082,34 @@ private function executeInserts($class)
 
                 $this->addToIdentityMap($entity);
             }
+        } else {
+            foreach ($theseInsertions as $oid => $entity) {
+                if(!isset($this->entityIdentifiers[$oid])) {
+                    //entity was not added to identity map because some identifiers are foreign keys to new entities.
+                    //add it now
+                    $identifier = [];
+                    $idFields   = $class->getIdentifierFieldNames();
+                    foreach ($idFields as $idField) {
```

Comment:
This block seems to detect whether an identifier is incomplete. Can it be moved to a `private` method?